projects/packages/videopress/src/client/admin/components/edit-video-details/use-edit-details.ts

```diff
@@ -61,47 +61,79 @@ export default () => {
 	const { data: video, isFetching } = useVideo( Number( videoId ) );
 
 	const [ updating, setUpdating ] = useState( false );
-	const [ data, setData ] = useState( video );
+
+	const [ data, setData ] = useState( {
+		title: video?.title,
+		description: video?.description,
+		caption: video?.caption,
+	} );
```

Comment:
Creating an internal state when these data are already stored in the state doesn't make sense. It's prone to get data inconsistency, adding unnecessary, complexity, etc.

Reply:
This `useEditDetails()` hook calls to the another `useVideo()` hook. A hook that was thought to edit video details calls to another hook with the same purpose. 

`L61` `const { data: video, isFetching } = useVideo( Number( videoId ) );`

You added three helpers to edit `title`, `description`, and `caption`...

```
	const setTitle = ( title: string ) => {
		updateData( { title } );
	};
	const setDescription = ( description: string ) => {
		updateData( { description } );
	};
	const setCaption = ( caption: string ) => {
		updateData( { caption } );
	};
```
... something that's already handled by the `useVideo()` hook.

These helpers also call to another hook named [useMetaUpdate()](https://github.com/Automattic/jetpack/blob/5496df912710d44964415eaac4af2a6afe841a3b/projects/packages/videopress/src/client/hooks/use-meta-update.js#L3).

`L17` `const updateMeta = useMetaUpdate( videoId );`

[useMetaUpdate()](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/client/hooks/use-meta-update.js#L9) performs the async request to hit the backend to update the data on the server side.

**The data handling implemented on these components is a mixture between internal state, custom hooks, and, and the usage of the data handling of the rest of the app. It doesn't look good to me, but it's your call.**

Reply:
Let's jump into the reasons.

If you take a look into the entire experience of the edit page, the user enters on it, edit the infos and hit a `Save changes` button, or the user could edit, and leave the pages (discarding the changes).

Based on this experience, we keep the data inside a local state, until the user hits the button, in that moment, we push the changes for backend and then for global state. With that, we do not mess with global state and don't need to discard or revert changes, since the local state gets prune as soon the user left the page.

For me, there's no reason to keep a transient data in global state, since the local one provides exactly what we want (discard it as soon the user left the page), if we start to use a global state for that, we'll need to prune it when the user leaves the page.

IMO, we should not make the global state a silver bullet, it doesn't fit for all situations, as everything.

Reply:
> If you take a look into the entire experience of the edit page, the user enters on it, edit the infos and hit a Save changes button, or the user could edit, and leave the page (discarding the changes).

Something 100% doable without adding an additional internal state.

> Based on this experience, we keep the data inside a local state (where the initial state is based on current video data), until the user hits the button, in that moment, we push the changes for backend and then for global state. With that, we do not mess with global state and don't need to discard or revert changes, since the local state gets prune as soon the user left the page.

This just adds unneeded complexity. It gets tricky event to explain it. 

You can take a look at the Video Privacy setting. The value is propagated to the state straightforwardly, hitting the backend. Everything happens in the State context (selector, resolvers, actions, and reducers. The functionallty is flawless IHMO.

https://user-images.githubusercontent.com/77539/194068029-72f4cd16-a906-4e0f-8720-e867b8f63890.mov


> For me, there's no reason to keep a transient data in global state ...

Clearly, it's a personal opinion but I would say quite incorrect. One of the benefits of using a global state provides the ability to propagate the data state among all components, on the fly, in a smootly way.

> ..., since the local one provides exactly what we want (discard it as soon the user left the page), if we start to use a global state for that, we'll need to prune it when the user leaves the page.

That is not correct. Everything flushes when the data is not propagated to the backend. The data is not persistent locally, so we shouldn't worry about it.

It's just a matter to implement it properly. The video above is a good example of how the data flow without issues.

Again, this implementation is almost disconnected from the global state, which makes it even more confusing from my PoV.

Reply:
> You can take a look at the Video Privacy setting. The value is propagated to the state straightforwardly, hitting the backend. Everything happens in the State context (selector, resolvers, actions, and reducers. The functionallty is flawless IHMO.

You're comparing two different features here, which makes me feel you don't fully understand this feature. In your example, as soon the user clicks on the button, the data needs to be propagated and pushed to backend, for sure this one don't need to implement a local state.

The experience on edit page is completely different, the user edit everything, and after, it clicks on `Save changes` and in this moment we should push the changes. If we start to push the changes directly to global state, if the user decides to not hit the save changes button, we'll need to revert the changes made into the global state.

> Everything flushes when the data is not propagated to the backend.

It flushes if you do a new fetch or refresh the page, if you're navigating inside the app, as expected in a SPA, the data still the same.

> One of the benefits of using a global state provides the ability to propagate the data state among all components.

Yes, but we should propagate truth data, not one that the user is editing and could discard until hits a button.